### PR TITLE
[FLINK-23632][docs] Fix the setup-pyflink-virtual-env.sh link

### PR DESCRIPTION
**Pull request by aajisaka**
## What is the purpose of the change

Fix the dead link to setup-pyflink-virtual-env.sh in https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/dev/python/faq/

JIRA: FLINK-23632

## Brief change log

Add back setup-pyflink-virtual-env.sh to fix dead link.

## Verifying this change

Manually built the document by running hugo locally. The script can be downloaded from the homepage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by aajisaka** [comment]
How can I proceed?

**Comment by haji08-ksd** [comment]
after flink-docs 1.12, no have downloads folder...
https://nightlies.apache.org/flink/flink-docs-release-1.12/downloads/setup-pyflink-virtual-env.sh

now no need setup-pyflink-virtual-env.sh ?

is it need to delete guide?

**Comment by gaborgsomogyi** [comment]
@aajisaka can you actualize this PR?
* Download the latest minikonda
* M1 support for Mac
* I think we can drop the 1.10 related stuff

**Comment by aajisaka** [comment]
Thank you @gaborgsomogyi 

The latest miniconda brings Python 3.11 and it's not supported even in latest Flink. Therefore I used Python 3.10 version of miniconda.

### Testing 

Ran the following command successfully on my M2 MacBook Pro.

```
./docs/static/downloads/setup-pyflink-virtual-env.sh 1.17.1
```

**Comment by gaborgsomogyi** [comment]
Tested on my Mac and works fine. Could you plz rebase to the latest master? I've getting the following error when trying to build docs:
```
Watching for changes in /Users/gaborsomogyi/flink/docs/{assets,content,content.zh,data,layouts,static,themes}
Watching for config changes in /Users/gaborsomogyi/flink/docs/config.toml
Start building sites … 
hugo v0.117.0-b2f0696cad918fb61420a6aff173eb36662b406e+extended darwin/arm64 BuildDate=2023-08-07T12:49:48Z VendorInfo=brew

Built in 107 ms
Error: error building site: assemble: "/Users/gaborsomogyi/flink/docs/content/docs/connectors/table/formats/debezium.md:240:1": failed to extract shortcode: shortcode "tabs" must be closed or self-closed
```

**Comment by gaborgsomogyi** [comment]
Most probably it will help to solve this issue too:
```
Aug 29 08:47:39 ERROR: tox 3.14.0 has requirement importlib-metadata<1,>=0.12; python_version < "3.8", but you'll have importlib-metadata 6.7.0 which is incompatible.
```

**Comment by gaborgsomogyi** [comment]
I've done the rebase locally and the doc generation and local serve on `localhost:1313` looks good, so this is the direction:
Screenshot 2023-08-29 at 12 14 01
Screenshot 2023-08-29 at 12 14 20

**Comment by aajisaka** [comment]
Thank you @gaborgsomogyi. Rebased to the latest master.

File: docs/static/downloads/setup-pyflink-virtual-env.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -e
+# download miniconda.sh
+if [[ `uname -s` == "Darwin" && `uname -m` == "arm64" ]]; then
+    wget "https://repo.anaconda.com/miniconda/Miniconda3-py310_23.5.2-0-MacOSX-arm64.sh" -O "miniconda.sh"
+elif [[ `uname -s` == "Darwin" ]]; then
+    wget "https://repo.anaconda.com/miniconda/Miniconda3-py310_23.5.2-0-MacOSX-x86_64.sh" -O "miniconda.sh"
```
Comment (gaborgsomogyi) [inline docs/static/downloads/setup-pyflink-virtual-env.sh:21]: This can be simplified to an embedded if:
```
if darwin
  if arm
  else
else
```

**Review by gaborgsomogyi — COMMENTED** [review]
When the last simplification is done then I think the code is good to go.

File: docs/static/downloads/setup-pyflink-virtual-env.sh (hunk 1 above)
Comment (aajisaka) [reply docs/static/downloads/setup-pyflink-virtual-env.sh]: Thanks. Fixed

**Review by gaborgsomogyi — APPROVED** [review]
LGTM. Unless there are any objections I will merge this tomorrow.

**Comment by aajisaka** [comment]
Thank you @gaborgsomogyi!

**Pull request merged**